source/common/http/conn_manager_utility.cc

```diff
@@ -122,18 +122,9 @@ void ConnectionManagerUtility::mutateRequestHeaders(
 
   // Generate x-request-id for all edge requests, or if there is none.
   if (config.generateRequestId() && (edge_request || !request_headers.RequestId())) {
-    std::string uuid = "";
-
-    try {
-      uuid = random.uuid();
-    } catch (const EnvoyException&) {
-      // We could not generate uuid, not a big deal.
-      config.stats().named_.failed_generate_uuid_.inc();
-    }
-
-    if (!uuid.empty()) {
-      request_headers.insertRequestId().value(uuid);
-    }
+    const std::string uuid = random.uuid();
+    ASSERT(!uuid.empty());
+    request_headers.insertRequestId().value(uuid);
```

Comment:
As an aside, you could make this even faster by writing the characters of the UUID into the header map directly (after calling something like a reserve() function or something) with your other change in place. Obviously not needed in this change, but feel free to add a TODO PERF if you feel like it.